diff --git a/drivers/net/mlx_sx/sx.h b/drivers/net/mlx_sx/sx.h
index 0cd05cc..a419320 100644
--- a/drivers/net/mlx_sx/sx.h
+++ b/drivers/net/mlx_sx/sx.h
@@ -52,8 +52,7 @@
 #include "counter.h"
 #include <linux/interrupt.h>
 #include <linux/version.h>
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(3, 13, 0) && (defined(RHEL_MAJOR) && defined(RHEL_MINOR) && RHEL_MAJOR == 7 && \
-    (RHEL_MINOR == 2 || RHEL_MINOR == 3)))
+#if 1
     #include <linux/timecounter.h>
 #endif
 #include <linux/clocksource.h>
