diff --git a/drivers/net/mlx_sx/catas.c b/drivers/net/mlx_sx/catas.c
index ca92f9b..ec47716 100644
--- a/drivers/net/mlx_sx/catas.c
+++ b/drivers/net/mlx_sx/catas.c
@@ -75,10 +75,10 @@ static void dump_err_buf(struct sx_dev *dev)
     }
 }
 
-static void poll_catas(unsigned long dev_ptr)
+static void poll_catas(struct timer_list *t)
 {
-    struct sx_dev  *dev = (struct sx_dev *)dev_ptr;
-    struct sx_priv *priv = sx_priv(dev);
+    struct sx_priv *priv = from_timer(priv, t, catas_err.timer);
+    struct sx_dev *dev = &priv->dev;
 
     if (__raw_readl(priv->catas_err.map)) {
         dump_err_buf(dev);
@@ -131,7 +131,7 @@ void sx_core_start_catas_poll(struct sx_dev *dev)
     }
 
     INIT_LIST_HEAD(&priv->catas_err.list);
-    init_timer(&priv->catas_err.timer);
+    timer_setup(&priv->catas_err.timer, poll_catas, 0);
     priv->catas_err.map = NULL;
 
     if (!priv->fw.catas_size) {
@@ -148,8 +148,6 @@ void sx_core_start_catas_poll(struct sx_dev *dev)
         return;
     }
 
-    priv->catas_err.timer.data = (unsigned long)dev;
-    priv->catas_err.timer.function = poll_catas;
     priv->catas_err.timer.expires =
         round_jiffies(jiffies + SX_CATAS_POLL_INTERVAL);
     add_timer(&priv->catas_err.timer);
