--- a/drivers/net/mlx_sx/sx_clock.c	2018-04-13 09:00:40.312290000 +0300
+++ b/drivers/net/mlx_sx/sx_clock.c	2018-04-13 09:08:17.547706000 +0300
@@ -50,6 +50,8 @@
 #define PTP_GC_TIME (5 * HZ) /* Event with age of more than 5 seconds should be garbage-collected */
 #define PTP_LATE_MATCH_TIME (HZ / 4) /* matching after 1/4 second or longer is 'late match' */

+typedef u64 cycle_t;
+
 extern struct sx_globals sx_glb;

 struct sx_priv *__priv = NULL;

