diff --git a/drivers/net/sx_netdev/sx_netdev_rtnetlink.c b/drivers/net/sx_netdev/sx_netdev_rtnetlink.c
index c7d9b1a196..5c8fff15aa 100644
--- a/drivers/net/sx_netdev/sx_netdev_rtnetlink.c
+++ b/drivers/net/sx_netdev/sx_netdev_rtnetlink.c
@@ -65,13 +65,14 @@ static void sx_netdev_setup(struct net_device *dev)
 
     memset(net_priv, 0, sizeof(*net_priv));
     ether_setup(dev);
+    dev->max_mtu = 0; /* do not limit the max MTU */
     dev->hard_header_len = ETH_HLEN + ISX_HDR_SIZE;
     dev->needs_free_netdev = true;
     net_priv->netdev = dev;
     INIT_DELAYED_WORK(&net_priv->pude_dwork, sx_netdev_dwork_func);
 }
 
-static int sx_netdev_validate(struct nlattr *tb[], struct nlattr *data[])
+static int sx_netdev_validate(struct nlattr *tb[], struct nlattr *data[], struct netlink_ext_ack *extack)
 {
     __u32    id;
     int      is_lag;
@@ -272,7 +273,8 @@ static int sx_netdev_port_mac_get(struct sx_dev *dev, u32 log_port, u64 *mac)
     return err;
 }
 
-static int sx_netdev_newlink(struct net *net, struct net_device *dev, struct nlattr *tb[], struct nlattr *data[])
+static int sx_netdev_newlink(struct net *net, struct net_device *dev, struct nlattr *tb[], struct nlattr *data[],
+                             struct netlink_ext_ack *extack)
 {
     struct sx_net_priv *net_priv = netdev_priv(dev);
     int                 swid = 0;
