diff --git a/drivers/net/sx_netdev/sx_bridge_rtnetlink.c b/drivers/net/sx_netdev/sx_bridge_rtnetlink.c
index 1c1a007..07cd168 100644
--- a/drivers/net/sx_netdev/sx_bridge_rtnetlink.c
+++ b/drivers/net/sx_netdev/sx_bridge_rtnetlink.c
@@ -67,7 +67,7 @@ static void sx_bridge_setup(struct net_device *dev)
     net_priv->netdev = dev;
 }
 
-static int sx_bridge_validate(struct nlattr *tb[], struct nlattr *data[])
+static int sx_bridge_validate(struct nlattr *tb[], struct nlattr *data[], struct netlink_ext_ack *extack)
 {
     __u16 id;
 
@@ -125,7 +125,8 @@ nla_put_failure:
     return -EMSGSIZE;
 }
 
-static int sx_bridge_newlink(struct net *net, struct net_device *dev, struct nlattr *tb[], struct nlattr *data[])
+static int sx_bridge_newlink(struct net *net, struct net_device *dev, struct nlattr *tb[], struct nlattr *data[],
+                             struct netlink_ext_ack *extack)
 {
     struct sx_net_priv *net_priv = netdev_priv(dev);
     int                 swid = 0;
