diff --git a/drivers/net/mlx_sx/debug/memtrack/mtrack.h b/drivers/net/mlx_sx/debug/memtrack/mtrack.h
index aae0bd3..2773155 100644
--- a/drivers/net/mlx_sx/debug/memtrack/mtrack.h
+++ b/drivers/net/mlx_sx/debug/memtrack/mtrack.h
@@ -50,6 +50,10 @@
 #include <linux/skbuff.h>       /* For skbufs handling */
 #include <asm/uaccess.h>    /* For copy from/to user */
 
+#undef ioremap
+#undef iounmap
+#undef ioremap_nocache
+
 #define MEMTRACK_ERROR_INJECTION_MESSAGE(file, line, func)                    \
     ({                                                                        \
          printk(KERN_ERR "%s failure injected at %s:%d\n", func, file, line); \
