diff --git a/drivers/net/sx_netdev/sx_bridge_rtnetlink.c b/drivers/net/sx_netdev/sx_bridge_rtnetlink.c
index 3fa95ce5f6..6a7c9813ce 100644
--- a/drivers/net/sx_netdev/sx_bridge_rtnetlink.c
+++ b/drivers/net/sx_netdev/sx_bridge_rtnetlink.c
@@ -64,6 +64,7 @@ static void sx_bridge_setup(struct net_device *dev)
     memset(net_priv, 0, sizeof(*net_priv));
     ether_setup(dev);
     dev->hard_header_len = ETH_HLEN + ISX_HDR_SIZE;
+    dev->needs_free_netdev = true;
     net_priv->netdev = dev;
 }
